### PR TITLE
[AJUN-438] Updated dependencies to polkadot v1.12.0

### PR DESCRIPTION
**Pull request by DidacSF**
## Description

Please describe your pull request and link relevant issues.

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [x] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [ ] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [ ] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [ ] Tests for the changes have been added
- [ ] Necessary documentation is added (if appropriate)
- [ ] Formatted with `cargo fmt --all`
- [ ] Linted with `cargo clippy --all-features --all-targets`
- [ ] Tested with `cargo test --workspace --all-features --all-targets`

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -840,7 +809,7 @@ dependencies = [
  "staging-xcm",
  "staging-xcm-builder",
  "staging-xcm-executor",
- "substrate-wasm-builder",
+ "substrate-wasm-builder 21.0.0",
```
Comment (clangenb) [inline Cargo.lock]: This is suspicious, we should not have the same dep twice here.

File: Cargo.lock [hunk 2]
```diff
@@ -2853,7 +2864,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "7d9ce6874da5d4415896cd45ffbc4d1cfc0c4f9c079427bd870742c30f2f65a9"
 dependencies = [
  "curve25519-dalek 4.1.2",
- "ed25519",
+ "ed25519 2.2.3",
```
Comment (clangenb) [inline Cargo.lock:2867]: Also surprising.

File: Cargo.lock [hunk 3]
```diff
@@ -10284,14 +10582,15 @@ dependencies = [
...
  "multihash-codetable",
  "parity-scale-codec",
  "prost 0.12.6",
- "prost-build",
- "rand",
+ "prost-build 0.12.6",
+ "rand 0.8.5",
  "sc-client-api",
  "sc-network",
+ "sc-network-types 0.11.0",
```
Comment (clangenb) [inline Cargo.lock:10593]: This is also a substrate dep that we shouldn't have twice.

File: Cargo.lock [hunk 4]
```diff
@@ -13189,6 +13582,26 @@ dependencies = [
  "wasm-opt",
 ]
 
+[[package]]
+name = "substrate-wasm-builder"
+version = "22.0.0"
```
Comment (clangenb) [inline Cargo.lock]: This is the new one we get through the update, we have to find out where we still have the old one.

File: runtime/bajun/src/xcm_config.rs [hunk 5]
```diff
@@ -427,7 +428,7 @@ impl cumulus_pallet_xcmp_queue::Config for Runtime {
 	type MaxInboundSuspended = sp_core::ConstU32<1_000>;
 	type ControllerOrigin = EnsureRoot<AccountId>;
 	type ControllerOriginConverter = XcmOriginToTransactDispatchOrigin;
-	type WeightInfo = cumulus_pallet_xcmp_queue::weights::SubstrateWeight<Runtime>;
+	type WeightInfo = weights::cumulus_pallet_xcmp_queue::WeightInfo<Runtime>;
```
Comment (clangenb) [inline runtime/bajun/src/xcm_config.rs:431]: Good catch 👍

**Review by clangenb — COMMENTED** [review]
Looks good to me in general! But we should find out where we have the duplicate deps from.

**Comment by DidacSF** [comment]
@clangenb Regarding the repeated deps, based on my investigation. The issue does seems to come from the Polkadot SDK itself...

```sh
sc-network-types v0.10.0
└── sc-network-common v0.39.0
    ├── cumulus-relay-chain-minimal-node v0.13.0
    ├── polkadot-gossip-support v13.0.0
    ├── polkadot-service v13.0.0
    ├── sc-consensus-grandpa v0.25.0
    ├── sc-informant v0.39.0
    ├── sc-network v0.40.0 # <---- Dependency set here
    ├── sc-network-gossip v0.40.0
    ├── sc-network-sync v0.39.0
    ├── sc-network-transactions v0.39.0
    ├── sc-offchain v35.0.0
    └── sc-service v0.41.0

sc-network-types v0.11.0
├── polkadot-node-jaeger v13.0.0
├── polkadot-node-network-protocol v13.0.0
├── polkadot-node-subsystem-types v13.0.0
├── sc-authority-discovery v0.40.0
├── sc-consensus v0.39.1
├── sc-consensus-beefy v19.0.0
├── sc-consensus-grandpa v0.25.0
├── sc-mixnet v0.10.0
├── sc-network v0.40.0 # <---- Also here?
├── sc-network-gossip v0.40.0
├── sc-network-light v0.39.0
├── sc-network-sync v0.39.0
├── sc-network-transactions v0.39.0
├── sc-offchain v35.0.0
└── sc-service v0.41.0
```

```sh
ed25519 v1.5.3
└── ed25519-dalek v1.0.1
    ├── litep2p v0.3.0
    │   └── sc-network-types v0.10.0 # <--- The duplication avobe measn we also get it here
    └── litep2p v0.4.0-rc.1
        ├── sc-network v0.40.0
        └── sc-network-types v0.11.0 # <--- Duplication from above
ed25519 v2.2.3
├── ed25519-dalek v2.1.1
│   ├── libp2p-identity v0.1.3
│   ├── sp-io v36.0.0
│   └── sp-statement-store v16.0.0
└── ed25519-zebra v4.0.3
    ├── smoldot v0.11.0
    └── sp-core v33.0.1
```

It seems the `sc-network` is not as well organized as other deps. **Not much we can do there unfortunately**, the good thing is that for the `substrate-wasm-builder` manually setting the dep in our `Cargo.toml` to `22.0.0` manages to eliminate the duplication at least.

**Review by clangenb — APPROVED** [review]
Alright, thanks a lot for the explanation. Ready to merge then IMO!

**Pull request merged**